### PR TITLE
wrap compositor in container

### PR DESCRIPTION
**Pull request by toddmedema**

### CODE REVIEW

File: app/components/Compositor.tsx [hunk 1]
```diff
@@ -131,81 +86,48 @@ export default class Main extends React.Component<MainProps, {}> {
...
+    if (this.props.settings.fontSize === 'SMALL') {
+      containerClass.push('smallFont');
+    } else if (this.props.settings.fontSize === 'LARGE') {
+      containerClass.push('largeFont');
     }
 
     return (
-      <div className={containerClass}>
+      <div className={containerClass.join(' ')}>
         <Provider store={getStore()}>
```
Comment (smartin015) [inline app/components/Compositor.tsx]: The Provider should be moved out to `Main.tsx` to align with typical react-redux ways (see https://redux.js.org/basics/usage-with-react#passing-the-store)

File: app/components/views/quest/cardtemplates/Template.tsx [hunk 2]
```diff
@@ -57,8 +55,9 @@ export function getCardTemplateTheme(card: CardState): CardThemeType {
     case 'NO_TIMER':
     case 'MID_COMBAT_ROLEPLAY':
       return 'dark';
+    // case 'ROLEPLAY':
```
Comment (smartin015) [inline app/components/views/quest/cardtemplates/Template.tsx]: Why not un-comment and have it fall through?

File: app/components/Compositor.tsx [hunk 3]
```diff
@@ -17,84 +31,31 @@ import SettingsContainer from './views/SettingsContainer'
...
-    if (!state.card || (this.state && newTS === this.state.ts && newKey === this.state.key)) {
-      return this.state;
-    }
+export default class Compositor extends React.Component<CompositorProps, {}> {
 
-    let card: JSX.Element;
-    let theme: CardThemeType = 'light';
-    switch(state.card.name) {
+  render() {
+    let card = <SplashScreenContainer/>;
```
Comment (smartin015) [inline app/components/Compositor.tsx]: Why initialize it with something when we have a strict `switch` statement directly after?

File: app/Main.tsx [hunk 4]
```diff
@@ -327,15 +327,15 @@ export function init() {
...
+  const CompositorContainer = require('./components/CompositorContainer').default;
   const base = getDocument().getElementById('react-app');
   if (!base) {
     throw new Error('Could not find react-app element');
   }
   ReactDOM.unmountComponentAtNode(base);
   ReactDOM.render(
     <MuiThemeProvider muiTheme={getMuiTheme(theme)}>
-      <Compositor/>
+      <CompositorContainer store={getStore()}/>
```
[the remark this answers is not in the document]
Comment (toddmedema) [reply app/Main.tsx]: Ah, good call!

File: app/components/Compositor.tsx (hunk 1 above)
Comment (toddmedema) [reply app/components/Compositor.tsx]: Done

File: app/components/views/quest/cardtemplates/Template.tsx (hunk 2 above)
Comment (toddmedema) [reply app/components/views/quest/cardtemplates/Template.tsx]: Linter won't let me

File: app/components/Compositor.tsx (hunk 3 above)
Comment (toddmedema) [reply app/components/Compositor.tsx]: Gives us free type initialization, primarily. But reduced this to `<span/>` to be clearer about that

File: app/components/Compositor.test.tsx [hunk 5]
```diff
@@ -1,5 +1,3 @@
-describe('Main', () => {
-  it('does not trigger animation if state update to same card');
-
-  it('intercepts and publishes touch/move events');
+describe('Compositor', () => {
+  it('Renders the provided card');
 });
```
[the remark this answers is not in the document]
Comment (toddmedema) [reply app/components/Compositor.test.tsx]: Added. Not sure what else to test here besides that it renders a provided card. Seems low value / high maintenance to write tests for each of the switch cases.

File: app/components/Compositor.test.tsx [hunk 6]
```diff
@@ -1,5 +1,3 @@
-describe('Main', () => {
-  it('does not trigger animation if state update to same card');
-
-  it('intercepts and publishes touch/move events');
+describe('Compositor', () => {
+  it('Renders the provided card');
```
[the remark this answers is not in the document]
Comment (toddmedema) [reply app/components/Compositor.test.tsx]: Done

**Pull request merged**